compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -2277,19 +2277,16 @@ declare_lint! {
     ///
     /// ### Explanation
     ///
-    /// Previous versions of Rust allowed function pointers and wide raw pointers in patterns.
-    /// While these work in many cases as expected by users, it is possible that due to
-    /// optimizations pointers are "not equal to themselves" or pointers to different functions
-    /// compare as equal during runtime. This is because LLVM optimizations can deduplicate
-    /// functions if their bodies are the same, thus also making pointers to these functions point
-    /// to the same location. Additionally functions may get duplicated if they are instantiated
+    /// Use of function pointers and wide raw pointers in patterns works in many cases as
```

Comment:
```suggestion
    /// Use of function pointers and `dyn Trait` raw pointers in patterns works in many cases as
```
I assume this doesn't fire on slice raw pointers?